apps/consumer-client/src/pages/pcdpass-examples/zk-eddsa-event-ticket-proof.tsx

```diff
@@ -0,0 +1,468 @@
+import { EdDSATicketPCDPackage } from "@pcd/eddsa-ticket-pcd";
+import {
+  constructPassportPcdGetRequestUrl,
+  openPassportPopup,
+  usePassportPopupMessages,
+  useSerializedPCD
+} from "@pcd/passport-interface";
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDPackage } from "@pcd/semaphore-identity-pcd";
+import { generateSnarkMessageHash } from "@pcd/util";
+import {
+  EdDSATicketFieldsToReveal,
+  ZKEdDSAEventTicketPCD,
+  ZKEdDSAEventTicketPCDArgs,
+  ZKEdDSAEventTicketPCDPackage
+} from "@pcd/zk-eddsa-event-ticket-pcd";
+import path from "path";
+import { useEffect, useMemo, useState } from "react";
+import { CodeLink, CollapsableCode, HomeLink } from "../../components/Core";
+import { ExampleContainer } from "../../components/ExamplePage";
+import { PCDPASS_URL } from "../../constants";
+
+/**
+ * Example page which shows how to use a Zuzalu-specific prove screen to
+ * request a Semaphore Group Membership PCD as a third party developer.
+ */
+export default function Page() {
+  const watermark = generateSnarkMessageHash(
+    "consumer-client zk-eddsa-event-ticket-pcd challenge"
+  );
+  const externalNullifier =
+    generateSnarkMessageHash("consumer-client").toString();
+
+  const [revealTicketId, setRevealTicketId] = useState(false);
+  const [revealEventId, setRevealEventId] = useState(true);
+  const [revealProductId, setRevealProductId] = useState(true);
+  const [revealTimestampConsumed, setRevealTimestampConsumed] = useState(false);
+  const [revealTimestampSigned, setRevealTimestampSigned] = useState(false);
+  const [revealAttendeeSemaphoreId, setRevealAttendeeSemaphoreId] =
+    useState(false);
+  const [revealIsConsumed, setRevealIsConsumed] = useState(false);
+  const [revealIsRevoked, setRevealIsRevoked] = useState(false);
+  const [validEventIdsInput, setValidEventIdsInput] = useState("");
+
+  const validEventIds = validEventIdsInput
+    .split(",")
+    .map((s) => s.trim())
+    .filter((s) => s != "");
+
+  const fieldsToReveal: EdDSATicketFieldsToReveal = useMemo(
```

Comment:
this `useMemo` isn't really going to be saving you much. If you care to use it, I recommend using it on `validEventIds`, `watermark`, and `externalNullifier` instead.